src/Profile/Magento2/Gateway/Local/Magento2LocalGateway.php

```diff
@@ -80,7 +80,8 @@ public function readEnvironmentInformation(MigrationContextInterface $migrationC
     {
         $connection = $this->connectionFactory->createDatabaseConnection($migrationContext);
         $profile = $migrationContext->getProfile();
-        if ($connection === null) {
+
+        if ($connection === null || !$connection->isConnected()) {
```

Comment:
I would remove this as pointed out in the other thread, as `isConnected` might return `false` at this point without checking if the connection actually works
```suggestion
        if ($connection === null) {
```